packages/components/src/Button/Button.js

```diff
@@ -1,30 +1,39 @@
 import React from 'react';
 import styled from 'styled-components';
 import PropTypes from 'prop-types';
+import { color, space, fontStyle } from 'styled-system';
+
+/**
+ * Things to decide / discuss (in small time box)
+ * 1) Typescript (maybe this makes spread ok)
+ *  a) if we can include this with styled-system then great
+ * 2) Testid / classname
+ */
 
 export const StyledButton = styled.button`
-  background: ${props => props.bgColor};
-  color: #fff;
-  border-radius: 4px;
+  ${color},
+  ${space},
+  ${fontStyle},
   cursor: pointer;
   font-size: 1rem;
   font-weight: 300;
-  padding: 9px 36px;
 `;
 
-const Button = ({ children, bgColor }) => (
-  <StyledButton bgColor={bgColor}>{children}</StyledButton>
-);
+// this is to ensure that the docs autogen from proptype
+// eslint-disable-next-line react/jsx-props-no-spreading
+const Button = props => <StyledButton {...props} />;
+
+// magic variants - buttons small secondary, large primary
 
 Button.propTypes = {
   /** takes any child */
   children: PropTypes.node.isRequired,
-  /** permitted background colors */
-  bgColor: PropTypes.oneOf(['red', 'blue', 'green'])
+  /** Button DOM type attribute */
+  type: PropTypes.oneOf(['button', 'submit'])
```

Comment:
```suggestion
  type: PropTypes.oneOf(['button', 'reset', 'submit'])
```

https://devdocs.io/html/element/button

Reply:
after doing some review i think we should maybe revisit the prop spreading thing because it looks like both material ui + ant spread